core/chains/evm/client/pool.go

```diff
@@ -76,54 +92,85 @@ func (p *Pool) Dial(ctx context.Context) error {
 	})
 }
 
-// dialRetryInterval controls how often we try to reconnect a dead node
-var dialRetryInterval = 5 * time.Second
+// nLiveNodes returns the number of currently alive nodes
+func (p *Pool) nLiveNodes() (nLiveNodes int) {
+	for _, n := range p.nodes {
+		if n.State() == NodeStateAlive {
+			nLiveNodes++
+		}
+	}
+	return
+}
 
 func (p *Pool) runLoop() {
 	defer p.wg.Done()
-	ticker := time.NewTicker(dialRetryInterval)
 
-	for {
-		select {
-		case <-p.chStop:
-			return
-		case <-ticker.C:
-			// re-dial all dead nodes
-			func() {
-				ctx, cancel := utils.ContextFromChan(p.chStop)
-				defer cancel()
-				ctx, cancel = context.WithTimeout(ctx, dialRetryInterval)
-				defer cancel()
-				// TODO: How does this play with automatic WS reconnects?
-				p.redialDeadNodes(ctx)
-			}()
-		}
+	// FIXME: Use config for this?
+	monitor := time.NewTicker(time.Second)
+	defer monitor.Stop()
+	// monitor := time.NewTicker(time.Minute)
+
+	type nodeWithState struct {
+		Node  string
+		State string
 	}
-}
 
-func (p *Pool) redialDeadNodes(ctx context.Context) {
-	for _, n := range p.nodes {
-		if n.State() == NodeStateDead {
-			if err := n.Dial(ctx); err != nil {
-				p.logger.Errorw(fmt.Sprintf("Failed to redial eth node: %v", err), "err", err, "node", n.String())
+	for {
+		select {
+		case <-monitor.C:
+			var total, dead int
+			counts := make(map[NodeState]int)
+			nodeStates := make([]nodeWithState, len(p.nodes))
+			for i, n := range p.nodes {
+				state := n.State()
+				nodeStates[i] = nodeWithState{n.String(), state.String()}
+				total++
+				if state != NodeStateAlive {
+					dead++
+				}
+				counts[state]++
 			}
-		}
-		if n.State() == NodeStateInvalidChainID || n.State() == NodeStateDialed {
-			if err := n.Verify(ctx, p.chainID); err != nil {
-				p.logger.Errorw(fmt.Sprintf("Failed to verify eth node: %v", err), "err", err, "node", n.String())
+			for _, state := range allNodeStates() {
+				count := counts[state]
+				promEVMPoolRPCNodeStates.WithLabelValues(p.chainID.String(), state.String()).Set(float64(count))
 			}
+
+			live := total - dead
+			p.logger.Tracew(fmt.Sprintf("Pool state: %d/%[1]d nodes live and %d/%[1]d nodes dead", live, total, dead), "nodeStates", nodeStates)
+			if total == dead {
+				p.logger.Criticalw(fmt.Sprintf("No EVM primary nodes available: 0/%d nodes are alive", total), "nodeStates", nodeStates)
+			} else if dead > 0 {
+				p.logger.Errorw(fmt.Sprintf("At least one EVM primary node is dead: %d/%d nodes are alive", live, total), "nodeStates", nodeStates)
+			}
+		case <-p.chStop:
+			return
 		}
 	}
 }
 
+// Close tears down the pool and closes all nodes
 func (p *Pool) Close() {
 	//nolint:errcheck
 	p.StopOnce("Pool", func() error {
 		close(p.chStop)
 		p.wg.Wait()
+
+		var closeWg sync.WaitGroup
+		closeWg.Add(len(p.nodes))
 		for _, n := range p.nodes {
-			n.Close()
+			go func(node Node) {
+				defer closeWg.Done()
+				node.Close()
+			}(n)
+		}
+		closeWg.Add(len(p.sendonlys))
```

Comment:
Need to do this add before spawning any of the goroutines in the preceding for.

Reply:
Should be ok to do it here, no? Since worst case the wg goes back down to zero then gets incremented again before we call Wait